cli/src/main/java/bio/terra/pfb/GitConfiguration.java

```diff
@@ -0,0 +1,40 @@
+package bio.terra.pfb;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+public class GitConfiguration {
+  private static final String GIT_PROPERTIES_FILE_NAME = "git.properties";
+  private static final String CLI_VERSION_BUILD_PROPERTY = "javapfb.cli.version.build";
+
+  public String getCliVersion() {
+    return readGitPropertiesValue(CLI_VERSION_BUILD_PROPERTY);
+  }
+
+  private String readGitPropertiesValue(String propertyName) {
+    String result = "";
+    InputStream inputStream = null;
+    try {
+      Properties prop = new Properties();
+      inputStream = getClass().getClassLoader().getResourceAsStream(GIT_PROPERTIES_FILE_NAME);
+      if (inputStream != null) {
+        prop.load(inputStream);
+      } else {
+        throw new FileNotFoundException(
+            "Property file '" + GIT_PROPERTIES_FILE_NAME + "' not found in the classpath");
+      }
+      return prop.getProperty(propertyName);
+    } catch (IOException e) {
+      System.out.println("Encountered exception while reading git properties: " + e);
+    } finally {
+      try {
+        inputStream.close();
+      } catch (IOException e) {
+        System.out.println(e);
+      }
+    }
+    return result;
+  }
```

Comment:
Even though you're not using spring, I think using jackson `ObjectMapper` would be a better way to do this. Add this dependency to `cli/build.gradle`:
```
    implementation 'com.fasterxml.jackson.core:jackson-databind:2.15.2'
```

and this code should work:
```
    record GitProperties(String gitHash, String gitTag, String build, String github) { }
    private final GitProperties gitProperties;

    public GitConfiguration() {
        gitProperties = loadProperties();
    }

    private GitProperties loadProperties() {
        try (var inputStream = getClass().getClassLoader().getResourceAsStream(GIT_PROPERTIES_FILE_NAME)) {
            return new ObjectMapper().readValue(inputStream, GitProperties.class);
        } catch (IOException e) {
            throw new RuntimeException("Failed to load git properties", e);
        }
    }

    public String getCliVersion() {
        return gitProperties.build;
    }
```

Reply:
It would be nice to keep the CLI as lightweight as possible and given that there is a reasonable way to do this without needing another dependency, I'm inclined to leave it as is. Your suggested code is definitely more concise -- are there are any other benefits to using the object mapper?

Reply:
It's true that adding this dependency will make the distribution bigger. If the cli is expected to be used by automation, then it will probably need to generate and consume JSON, which the library could be used for. If however you don't think you'll need to read or produce JSON at all then it may not be worth it.

Reply:
If you don't want to use jackson, here's how I'd write it using `Properties`.
- If you open the `InputStream` inside a `try` then you don't need to call `close()`
- I wouldn't bother checking for `null` explicitly since `Properties.load()` already does that
- It looks like picocli has its own set of exceptions which are probably better than using `RuntimeException`, which sonar will probably complain about.

```
  private final Properties properties;

  public GitConfiguration() {
    properties = loadProperties();
  }

  private Properties loadProperties() {
    try (var inputStream =
        getClass().getClassLoader().getResourceAsStream(GIT_PROPERTIES_FILE_NAME)) {
      var props = new Properties();
      props.load(inputStream);
      return props;
    } catch (IOException e) {
      throw new CommandLine.InitializationException("Failed to load git properties", e);
    }
  }

  public String getCliVersion() {
    return properties.getProperty(CLI_VERSION_BUILD_PROPERTY);
  }
```

Reply:
That's a good point - I probably will end up needing jackson for the actual work required in this library. (TBD if needed in the CLI). 
In the mean time, I'll look at this alternate you've provided tomorrow.

Reply:
This is great, thank you. I also wasn't totally sure how to handle errors - Thanks for this example of using picocli's exception mechanism.